modules/schematics/src/entity/files/__name@dasherize@if-flat__/__name@dasherize@group-reducers__.reducer.ts.template

```diff
@@ -15,43 +15,51 @@ export const initialState: State = adapter.getInitialState({
-  ),
-  on(<%= classify(name) %>Actions.add<%= classify(name) %>s,
-    (state, action) => adapter.addMany(action.<%= camelize(name) %>s, state)
-  ),
-  on(<%= classify(name) %>Actions.upsert<%= classify(name) %>s,
-    (state, action) => adapter.upsertMany(action.<%= camelize(name) %>s, state)
-  ),
-  on(<%= classify(name) %>Actions.update<%= classify(name) %>,
-    (state, action) => adapter.updateOne(action.<%= camelize(name) %>, state)
-  ),
-  on(<%= classify(name) %>Actions.update<%= classify(name) %>s,
-    (state, action) => adapter.updateMany(action.<%= camelize(name) %>s, state)
-  ),
-  on(<%= classify(name) %>Actions.delete<%= classify(name) %>,
-    (state, action) => adapter.removeOne(action.id, state)
-  ),
-  on(<%= classify(name) %>Actions.delete<%= classify(name) %>s,
-    (state, action) => adapter.removeMany(action.ids, state)
-  ),
-  on(<%= classify(name) %>Actions.load<%= classify(name) %>s,
-    (state, action) => adapter.setAll(action.<%= camelize(name) %>s, state)
-  ),
-  on(<%= classify(name) %>Actions.clear<%= classify(name) %>s,
-    state => adapter.removeAll(state)
-  ),
-);
+export const <%= pluralize(name) %>Feature = createFeature({
+  name: <%= pluralize(name) %>FeatureKey,
+  reducer: createReducer(
+    initialState,
+    on(<%= classify(name) %>Actions.add<%= classify(name) %>,
+      (state, action) => adapter.addOne(action.<%= camelize(name) %>, state)
+    ),
+    on(<%= classify(name) %>Actions.upsert<%= classify(name) %>,
+      (state, action) => adapter.upsertOne(action.<%= camelize(name) %>, state)
+    ),
+    on(<%= classify(name) %>Actions.add<%= classify(name) %>s,
+      (state, action) => adapter.addMany(action.<%= camelize(name) %>s, state)
+    ),
+    on(<%= classify(name) %>Actions.upsert<%= classify(name) %>s,
+      (state, action) => adapter.upsertMany(action.<%= camelize(name) %>s, state)
+    ),
+    on(<%= classify(name) %>Actions.update<%= classify(name) %>,
+      (state, action) => adapter.updateOne(action.<%= camelize(name) %>, state)
+    ),
+    on(<%= classify(name) %>Actions.update<%= classify(name) %>s,
+      (state, action) => adapter.updateMany(action.<%= camelize(name) %>s, state)
+    ),
+    on(<%= classify(name) %>Actions.delete<%= classify(name) %>,
+      (state, action) => adapter.removeOne(action.id, state)
+    ),
+    on(<%= classify(name) %>Actions.delete<%= classify(name) %>s,
+      (state, action) => adapter.removeMany(action.ids, state)
+    ),
+    on(<%= classify(name) %>Actions.load<%= classify(name) %>s,
+      (state, action) => adapter.setAll(action.<%= camelize(name) %>s, state)
+    ),
+    on(<%= classify(name) %>Actions.clear<%= classify(name) %>s,
+      state => adapter.removeAll(state)
+    ),
+  )
+  extraSelectors: ({ select<%= capitalize(pluralize(name)) %>State }) => ({
+    ...adapter.getSelectors(select<%= capitalize(pluralize(name)) %>State)
+  }),
+});
 
 export const {
+  name,
+  reducer,
```

Comment:
We don't need to export `name`, because `<%= pluralize(name) %>FeatureKey` constant is already exported. Also, we can keep reducer creation as before to avoid additional nesting. So we can remove `name` and `reducer` from here:

```suggestion
```

`reducer` and feature object creation will look like this:

```
// keep reducer as it is
export const reducer = createReducer(
  /* ... */
);

// `createFeature` usage will look like this:
export const <%= pluralize(name) %>Feature = createFeature({
  name: <%= pluralize(name) %>FeatureKey,
  reducer,
  extraSelectors: ({ select<%= capitalize(pluralize(name)) %>State }) => ({
    ...adapter.getSelectors(select<%= capitalize(pluralize(name)) %>State)
  }),
});
```